lib/data_cleanup/rules/section/fix_duplicate_number.rb

```diff
@@ -10,18 +10,74 @@ def description
         end
 
         def call
-          data = ::Section.group(:number, :phase_id)
-                          .count
-                          .select { |k,v| v > 1 }
-          data.each do |values, count|
-            number, phase_id = *values
-            ids = ::Section.where(phase_id: phase_id)
-                           .order("number ASC, created_at ASC")
-                           .pluck(:id)
-            phase = ::Phase.find(phase_id)
+          # A set of unique phase_ids that contain Sections with duplicate numbers
+          phase_ids = ::Section.group(:number, :phase_id)
+                               .count
+                               .select { |k,v| v > 1 }
+                               .collect { |values, count| values.last }
+                               .uniq
+
+          phase_ids.each do |phase_id|
+            log("Re-setting number order for Sections in Phase##{phase_id}")
+            phase    = ::Phase.find(phase_id)
+            sections = phase.sections
```

Comment:
Can you not just do `::Phase.includes(:sections).find(phase_id)` since you're referencing `.sections` again within the update methods?

Reply:
Using `includes()` would run 2 DB queries:
1. Load the Phase with given `id`
2. Load all Sections with given `phase_id`

In this case, it's the same as doing:

1. `@phase = Phase.find(id)`
2. `@sections = @phase.sections`

Unless I'm missing something?